.github/workflows/ci.yml

```diff
@@ -174,78 +174,45 @@ jobs:
       run: |
         brew install automake
 
-    - name: Set up JDK 18
-      uses: actions/setup-java@v3
-      with:
-        distribution: 'zulu'
-        java-version: 18
-
-    - name: Setup Android SDK
-      uses: android-actions/setup-android@v2
-
-    - name: Gradle cache
-      uses: gradle/gradle-build-action@v2
-      with:
-        gradle-version: 7.6.1
-
-    - name: Build Socket Runtime
+    - name: Android FTE
       run: |
-        yes | sdkmanager --licenses > /dev/null
-        sdkmanager --install 'ndk;25.0.8775105' 'build-tools;33.0.2' 'system-images;android-33;google_apis;x86_64'
-        VERBOSE=1 ./bin/install.sh
-        ./bin/ci_version_check.sh
+        ./bin/android-functions.sh --android-fte --yes-deps
       env:
+        VERBOSE: 1
+        DEBUG: 1
+        NO_IOS: 1
         CI: true
         SSC_ANDROID_CI: true
+        ANDROID_HOME: 
+        ANDROID_SUPPORTED_ABIS: x86_64
+        JAVA_HOME: ""
         NODE_AUTH_TOKEN: ${{ secrets.PACKAGE_PAT }}
 
-    - name: AVD cache
-      uses: actions/cache@v3
-      id: avd-cache
-      with:
-        path: |
-          ~/.android/avd/*
-          ~/.android/adb*
-        key: avd-33
-
-    - name: create AVD and generate snapshot for caching
-      if: steps.avd-cache.outputs.cache-hit != 'true'
-      uses: reactivecircus/android-emulator-runner@v2
-      with:
-        api-level: 33
-        arch: x86_64
-        disable-animations: false
-        emulator-boot-timeout: 1200
-        emulator-options: -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
-        force-avd-creation: false
-        ndk: 25.0.8775105
-        profile: pixel_5
-        target: google_apis
-        script: echo "Generated AVD snapshot for caching."
+    - name: Build Socket Runtime
+      run: |
+        ./bin/install.sh
+        ./bin/ci_version_check.sh
       env:
+        VERBOSE: 1
+        NO_IOS: true
         CI: true
         SSC_ANDROID_CI: true
+        ANDROID_HOME: 
+        ANDROID_SUPPORTED_ABIS: x86_64
         NODE_AUTH_TOKEN: ${{ secrets.PACKAGE_PAT }}
 
     - name: Run emulator tests
-      uses: reactivecircus/android-emulator-runner@v2
-      with:
-        api-level: 33
-        arch: x86_64
-        disable-animations: false
-        emulator-boot-timeout: 1200
-        emulator-options: -no-snapshot-save -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
-        force-avd-creation: false
-        ndk: 25.0.8775105
-        profile: pixel_5
-        target: google_apis
-        script: |
-          ./bin/ci_version_check.sh
+      run: |
           npm install
-          ANDROID_SUPPORTED_ABIS="x86_64" npm run test:android
+          npm run test:android-emulator
       env:
-        CI: true
+        VERBOSE: 1
+        NO_IOS: 1
+        CI: 1
         SSC_ANDROID_CI: true
+        ANDROID_SUPPORTED_ABIS: x86_64
+        JAVA_HOME: ""
```

Comment:
hmm... maybe we need some comment here

Reply:
Have added comments. 
These are definitely required. We need to override the CI provided values.
We want to use the `.ssc.env` values, but env vars take precedence, in this case that is bad.